### PR TITLE
Support for Fabric 1.17.1

### PR DESCRIPTION
**Pull request by belak**
This is best reviewed commit-by-commit (I generally rebase to keep the history clean anyway). The first commit copies the fabric-1.17 folder, updates the code to reference 1.17.1 rather than 1.17 and adds the new project to the build. Everything after that is fixes to get this to build/work.

### CODE REVIEW

File: fabric-1.17.1/src/main/resources/fabric.mod.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    ]
+  },
+  "mixins": [
+    "dynmap.mixins.json"
+  ],
+
+  "depends": {
+    "fabricloader": ">=0.7.4",
+    "fabric": ">=0.17.0",
+    "minecraft": "1.17"
```
Comment (emmatyping) [inline fabric-1.17.1/src/main/resources/fabric.mod.json]:
```suggestion
    "minecraft": "1.17.1"
```

This is required for it to be loaded.

Comment (belak) [reply fabric-1.17.1/src/main/resources/fabric.mod.json]: Good catch

File: fabric-1.17.1/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# Fabric Properties
+# check these on https://fabricmc.net/use
+minecraft_version=1.17.1
+yarn_mappings=1.17.1+build.1
+loader_version=0.11.6
+# Mod Properties
+archives_base_name=Dynmap
+# Dependencies
+# currently not on the main fabric site, check on the maven: https://maven.fabricmc.net/net/fabricmc/fabric-api/fabric-api
+fabric_version=0.36.1+1.17
```
Comment (mikeprimm-triva) [inline fabric-1.17.1/gradle.properties:12]: I think this needs to be fixed - please confirm

**Review by mikeprimm-triva — CHANGES_REQUESTED** [review]
I think there is a problem with the fabric version in the build settings - please confirm

**Comment by CCF100** [comment]
I built this port and attempted to do a fullrender of my map:
image
(I have the caves and cliffs feature datapack installed, which might be what's causing this)
Update: #3419

File: fabric-1.17.1/gradle.properties (hunk 2 above)
Comment (belak) [reply fabric-1.17.1/gradle.properties:12]: Nope, this is correct. Thanks for the callout though.

See https://maven.fabricmc.net/net/fabricmc/fabric-api/fabric-api/0.36.1%2B1.17/ and https://www.curseforge.com/minecraft/mc-mods/fabric-api/files/3377591

**Comment by JulienTant** [comment]
Anything needs change / I can help with?

**Comment by mikeprimm** [comment]
Giving it a go - thanks!!

**Comment by SimisFul** [comment]
Thanks for the continued hard work everyone, us end users really appreciate it :)

**Comment by mikeprimm** [comment]
If anyone with Fabric 1.17.1 has a chance, please check out the development build - http://dynmap.us/builds/dynmap/Dynmap-3.2-SNAPSHOT-fabric-1.17.1.jar

**Comment by kosma** [comment]
Been running the dev build for a day, so far no issues.

**Comment by JulienTant** [comment]
same, been running for a few days now everything seems to work

**Comment by elhertz** [comment]
Tried it out with the "Better Minecraft [FABRIC]" server modpack for 1.17.1 (available on curseforge) and it doesn't work
"**[Dynmap] Exception while loading chunks
java.lang.ArrayIndexOutOfBoundsException: Index -2 out of bounds for length 17**"

Map loads like this https://i.imgur.com/kD4jp9Q.png
This is the full log http://hastebin.com/jarikobedi.properties

**Pull request merged**